### PR TITLE
Captcha cleanup

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue # .

### Summary of Changes
This PR cleans up Captcha usage.

When Captcha plugin is set in Global or Component configuration, disabling, uninstalling or changing access level of the plugin causes system messages and fatal errors. Generally, plugin status should not cause such issues.

### Testing Instructions

Enable and configure a Captcha plugin.
Select the plugin in Global Configuration and in component (com_content and com_contact) options.
Disable or uninstall the plugin.

a) Enter username reminder or password reset view in Users component.
b) Submit the form in one of the above views.
c) Submit Contact form.
d) Submit Article from frontend.
e) Now enable the plugin (or install and enable if it was uninstalled) and repeat the above scenarios.

### Expected result

a) No errors. Captcha is not shown in form.
b, c, d) Forms successfully submitted.
e) Works like before patch.

### Actual result

a) System message:
>Error
Captcha plugin not set or not found. Please contact a site administrator.

Captcha field label is shown in form.

b, c, d)
Submitting form causes error:
>Error
Call to a member function checkAnswer() on null 

e) Works like before patch.

### Documentation Changes Required
No.

### CODE REVIEW

File: libraries/src/Form/Rule/CaptchaRule.php [hunk 1]
```diff
@@ -47,16 +47,21 @@ public function test(\SimpleXMLElement $element, $value, $group = null, Registry
...
-		if ($plugin === 0 || $plugin === '0')
+		// Plugin not set
+		if (!$plugin)
 		{
 			return true;
 		}
 		else
 		{
-			$captcha = Captcha::getInstance((string) $plugin, array('namespace' => (string) $namespace));
+			$namespace = $element['namespace'] ?: $form->getName();
```
Comment (Quy) [inline libraries/src/Form/Rule/CaptchaRule.php]: Lines 57-58 don't have to be in the else statement. Remove `else { }`.

Comment (SharkyKZ) [reply libraries/src/Form/Rule/CaptchaRule.php]: Done. Thanks.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 3a2703abe2c5f0f9f3c436135af923f988e4f6b5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20288">issues.joomla.org/tracker/joomla-cms/20288</a>.</sub>

**Comment by PhilETaylor** [comment]
@joomla/security

File: components/com_contact/views/contact/view.html.php [hunk 2]
```diff
@@ -52,8 +52,9 @@ class ContactViewContact extends JViewLegacy
 	/**
 	 * Should we show a captcha form for the submission of the contact request?
 	 *
-	 * @var   bool
-	 * @since 3.6.3
+	 * @var         bool
```
Comment (Quy) [inline components/com_contact/views/contact/view.html.php:55]: Spell out `bool`. Repeat in other file.

**Comment by twister65** [comment]
I have tested this item :white_check_mark: successfully on 3a2703abe2c5f0f9f3c436135af923f988e4f6b5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20288">issues.joomla.org/tracker/joomla-cms/20288</a>.</sub>

**Comment by SharkyKZ** [comment]
Thanks but I'm closing this due to conflicts.

**Pull request closed without merging**